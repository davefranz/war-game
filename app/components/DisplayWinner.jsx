import React from 'react';

// this is a little small to be its own component
const DisplayWinner = ({ winner }) => {
  if (!winner) return <p id="winner">Click Battle to Begin</p>;
  else if (winner.length > 1) return <p id="winner">{winner}!</p>;
  else return <p id="winner">PLAYER {winner} WINS!</p>; 
}

export default DisplayWinner;