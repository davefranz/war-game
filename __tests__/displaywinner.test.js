import React from "react";
import DisplayWinner from '../app/components/DisplayWinner.jsx';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';


describe('display winner componen tests', () => {
  it("renders", () => { 
    const { asFragment } = render(<DisplayWinner winner="Barbara" />);
    expect(asFragment()).toMatchSnapshot();
  });
})