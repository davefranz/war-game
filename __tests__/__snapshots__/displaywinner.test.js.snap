// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display winner componen tests renders 1`] = `
<DocumentFragment>
  <p
    id="winner"
  >
    Barbara!
  </p>
</DocumentFragment>
`;
